### PR TITLE
Feature/sale details

### PR DESCRIPTION
**Pull request by Walterieux**
I modified the tests to achieve 100% coverage. 
A message appear when you click on "contact the seller" : "This functionality is not implemented yet". 
Image information can be passed through the Intent when activity is called.

### CODE REVIEW

**Comment by armen-h** [comment]
Hi,
1. in test: contactTheSellerNotImplementedYet(), have you tried the code from the bootcamp? Not sure that you have to go through a thread:

@Test
    public void testCanGreetUsers() {
        onView(withId(R.id.mainName)).perform(typeText("from my unit test")).perform(closeSoftKeyboard());
        onView(withId(R.id.mainGoButton)).perform(click());
        onView(withId(R.id.greetingMessage)).check(matches(withText("Hello from my unit test!")));
    }

2. If you use the same values redundantly (aka constants, for example for text size, margins, textbox width/length), I suggest you declare them in res/values as constants

**Comment by Walterieux** [comment]
> Hi,
> 
>     1. in test: contactTheSellerNotImplementedYet(), have you tried the code from the bootcamp? Not sure that you have to go through a thread:
> 
> 
> @test
> public void testCanGreetUsers() {
> onView(withId(R.id.mainName)).perform(typeText("from my unit test")).perform(closeSoftKeyboard());
> onView(withId(R.id.mainGoButton)).perform(click());
> onView(withId(R.id.greetingMessage)).check(matches(withText("Hello from my unit test!")));
> }

Yes, i tried it first but i got this error : 
androidx.test.espresso.PerformException: Error performing 'single click' ...
see https://stackoverflow.com/questions/29786413/performexception-error-performing-single-click for more information.

> 
>     1. If you use the same values redundantly (aka constants, for example for text size, margins, textbox width/length), I suggest you declare them in res/values as constants

Good idea, I will do this before merging.

File: app/src/main/java/ch/epfl/polybazaar/SaleDetails.java [hunk 1]
```diff
@@ -19,11 +19,14 @@ protected void onCreate(Bundle savedInstanceState) throws NullPointerException{
 
         Bundle b = getIntent().getExtras();
         if(b == null)
-            throw new NullPointerException();
+            b = new Bundle();
```
Comment (JanMaxime) [inline app/src/main/java/ch/epfl/polybazaar/SaleDetails.java]: Could you rename that variable with a more explicit name ?

Comment (Walterieux) [reply app/src/main/java/ch/epfl/polybazaar/SaleDetails.java]: > Could you rename that variable with a more explicit name ?

Done

**Pull request merged**